src/ConformalTracking.cc

```diff
@@ -1667,6 +1667,371 @@ void ConformalTracking::extendHighPT(UniqueKDTracks& conformalTracks, SharedKDCl
+                                << deltaChi2zs << std::endl;
+          tempTrack->remove(tempTrack->m_clusters.size());
+          streamlog_out(DEBUG9) << "-- tempTrack has now " << tempTrack->m_clusters.size() << " hits " << std::endl;
+
+          double chi2cut = parameters._chi2cut;
+          if (deltaChi2 > chi2cut || deltaChi2zs > chi2cut) {
+            streamlog_out(DEBUG9) << "-- killed by chi2 cut" << std::endl;
+            continue;
+          }
+          streamlog_out(DEBUG9) << "-- valid candidate" << std::endl;
+
+          bestClustersWithChi2[nhit] = deltaChi2;
+
+          // bestCluster still empty - fill it with the first candidate
+          if (bestCluster == NULL) {
+            bestCluster = nhit;
+            bestChi2    = deltaChi2;
+            //	    streamlog_out(DEBUG9) << "-- First candidate: [x,y] = [" << bestCluster->getX() << ", " << bestCluster->getY() << "], deltachi2 = " << bestChi2 << std::endl;
+          }
+          // bestCluster contains already a candidate
+          else {
+            if (deltaChi2 < bestChi2) {
+              bestCluster = nhit;
+              bestChi2    = deltaChi2;
+              //	      streamlog_out(DEBUG9) << "-- Best cluster updated: [x,y] = [" << bestCluster->getX() << ", " << bestCluster->getY() << "], deltachi2 = " << bestChi2 << std::endl;
+
+            } else {
+              //	      streamlog_out(DEBUG9) << "-- Best cluster unchanged: [x,y] = [" << bestCluster->getX() << ", " << bestCluster->getY() << "], deltachi2 = " << bestChi2 << std::endl;
+              continue;
+            }
+          }
+
+        }  // end if on the extension layer
+
+      }  // end loop on neighbours
+
+      streamlog_out(DEBUG9) << "-- this seed cells has " << bestClustersWithChi2.size() << " good candidates." << std::endl;
+
+      //put the best cluster already found
+      if (bestCluster != NULL) {
+        bestClusters.push_back(bestCluster);
+      }
+
+      //put all the other cluster with a similar chi2 - probably duplicates
+      float chi2window = 100.0;
+      if (bestClustersWithChi2.size() > 0) {
+        for (auto clu : bestClustersWithChi2) {
+          streamlog_out(DEBUG9) << "-- Best cluster candidate: [x,y] = [" << clu.first->getX() << ", " << clu.first->getY()
+                                << "]; r = " << clu.first->getR() << "; radius = " << clu.first->getRadius() << std::endl;
+          streamlog_out(DEBUG9) << "-- chi " << clu.second << " compared to best chi " << bestChi2 << " of best candidate."
+                                << std::endl;
+          if (clu.second < bestChi2 + chi2window && clu.second > bestChi2 - chi2window) {
+            if (clu.first->sameSensor(bestCluster))
+              streamlog_out(DEBUG9) << "-- current candidate and best one are in the same sensor " << std::endl;
+            if (bestCluster != NULL && !clu.first->sameSensor(bestCluster))
```

Comment:
In fact the condition bestCluster != NULL is not needed here, since we are in the if(bestClustersWithChi2.size() > 0), thus we have at least one bestCluster.
So line 1948 can just become an 'else'